Latest push to main caused CI error related to Sonar scanner

Below is the error message from the CI run:

```
Error:  No plugin found for prefix 'sonar' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (/home/runner/.m2/repository), central (https://repo.maven.apache.org/maven2)] -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/NoPluginFoundForPrefixException
Error: Process completed with exit code 1.
```

Here is a related issue with the same error: https://community.sonarsource.com/t/fixing-no-plugin-found-for-prefix-sonar/152933

It doesn't look like there is a dependency in the POM file for the sonar build plugin.  Perhaps adding an entry will fix the issue.

Note that [SPDX Java Core POM file](https://github.com/spdx/spdx-java-core/blob/main/pom.xml) does have a sonar plugin entry.

--- a/pom.xml
+++ b/pom.xml
@@ -329,6 +329,11 @@
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
+				<groupId>org.sonarsource.scanner.maven</groupId>
+				<artifactId>sonar-maven-plugin</artifactId>
+				<version>5.5.0.6356</version>
+			</plugin>
 		</plugins>
 	</build>
 	<reporting>

